packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,248 @@
+      details: UniversalDetailsStruct,
+      abis: optional(any()),
+    }),
+    AuthorizableStruct,
+    BaseRequestStruct,
+  ),
+  'ExecuteTxnRequestStruct',
+  (value) => {
+    if (value.invocations.length === 0) {
+      return 'Invocations cannot be empty';
+    }
+    for (const invocation of value.invocations) {
+      if (invocation.type !== TransactionType.INVOKE) {
+        return `Invocations should be of type ${TransactionType.INVOKE} received ${invocation.type}`;
+      }
+    }
+    return true;
+  },
+);
+
+export const ExecuteTxnResponseStruct = object({
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  transaction_hash: string(),
+});
+
+export type ExecuteTxnParams = Infer<typeof ExecuteTxnRequestStruct> & Json;
+
+export type ExecuteTxnResponse = Infer<typeof ExecuteTxnResponseStruct>;
+
+const getInvokeCalls = (invocations: Invocations): Call[] => {
+  return invocations
+    .filter((invocation) => invocation.type === TransactionType.INVOKE)
+    .map((invocation: any) => invocation.payload as Call);
+};
+
+/**
+ * The RPC handler to execute a transaction.
+ */
+export class ExecuteTxnRpc extends AccountRpcController<
+  ExecuteTxnParams,
+  ExecuteTxnResponse
+> {
+  protected requestStruct = ExecuteTxnRequestStruct;
+
+  protected responseStruct = ExecuteTxnResponseStruct;
+
+  /**
+   * Execute the transaction request handler.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.invocations - The invocations to execute (only invocations of type TransactionType.INVOKE)
+   * @param params.abis - The abis associated to invocations.
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, invocations, abis, details, enableAuthorize } = params;
+
+    const calls = getInvokeCalls(invocations as Invocations);
+
+    const accountDeployed = await isAccountDeployed(this.network, address);
+
+    if (!accountDeployed) {
+      const { callData } = getAccContractAddressAndCallData(
+        this.account.publicKey,
+      );
+
+      const deployAccountpayload = {
+        classHash: ACCOUNT_CLASS_HASH,
+        contractAddress: address,
+        constructorCalldata: callData,
+        addressSalt: this.account.publicKey,
+      };
+
+      invocations.unshift({
+        type: TransactionType.DEPLOY_ACCOUNT,
+        payload: deployAccountpayload,
+      });
+    }
+
+    if (
+      // Get Starknet expected not to show the confirm dialog, therefore,
+      // `enableAuthorize` will set to false to bypass the confirmation
+      // TODO: enableAuthorize should set default to true
+      enableAuthorize &&
```

Comment:
execute txn always require authorize, so no need this arg

Reply:
enableAuthorize is used to bypass confirmation for the get-starknet use case no ?

Reply:
yes, but execute txn always need dialog